### ISSUE TITLE
defmt-test: #[ignore] attribute support

### ISSUE THREAD

**Issue by japaric** [knurling-rs/defmt#390]
Support this

``` rust
// inside the `#[defmt_test::test]` module
#[ignore] 
#[test]
fn foo() {}
```

Tests marked as `ignore`-d should be compiled but not executed.
Ignored tests should be reported on the console output but should not be account for in the progress counter (e.g. `1/3`).

**Comment by Urhengulas** [comment]
* Clarification question: The test should get compiled, but never executed. What will then get reported in the console?
* Also is compatibility with the `--include-ignored` flag desired? ([link](https://doc.rust-lang.org/reference/attributes/testing.html#the-ignore-attribute))

**Comment by japaric** [comment]
> Clarification question: The test should get compiled, but never executed. What will then get reported in the console?

Ideally, something similar to what the built-in test harness reports: 
``` console
test tests::it_works ... ignored
```

> Also is compatibility with the --include-ignored flag desired?

we don't support "test" related flags in probe-run. That's blocked on [this RFC](https://github.com/knurling-rs/probe-run/issues/148)

**Comment by japaric** [comment]
implemented in #618
